scalameter-core/src/main/scala/org/scalameter/InvocationCounting.scala

```diff
@@ -0,0 +1,64 @@
+package org.scalameter
+
+import java.io.File
+import scala.collection.Seq
+import org.scalameter.execution.invocation.InvocationCountMatcher
+import org.scalameter.execution.invocation.instrumentation.{Instrumentation, MethodInvocationCounter}
+import org.scalameter.Key._
+
+
+/** Mixin for all [[org.scalameter.Measurer]] implementations that perform any kind of method invocation counting. */
+trait InvocationCounting extends Measurer {
+  def matcher: InvocationCountMatcher
+
+  def units: String = "#"
+
+  def measure[T, U](context: Context, measurements: Int, setup: (T) => Any, tear: (T) => Any, regen: () => T, snippet: (T) => Any): Seq[Double] = {
+    var iteration = 0
+    val invocationsBuilder = List.newBuilder[Double]
+    var value: T = null.asInstanceOf[T]
+    var obj: Any = null.asInstanceOf[Any]
+    val numMethods = context.goe(exec.measurers.methodInvocationLookupTable,
+      sys.error("Measurer.prepareContext must be called before Measurer.measure")).length
+
+    while (iteration < measurements) {
```

Comment:
This measurer is free to ignore the `measurements` argument, since we expect that it will be invoked for deterministic runs, where the invocation counts are constant.

Perhaps you can introduce a flag `Key.exec.assumeDeterministicRun` or something like that, which defaults to `true`, and assumes that `measurements` can be ignored.